### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by blakerosenthal**
Doc fixes!

- Fix failing doc builds: update environment and remove/fix outdated refs in RST source files
- Add intake-dremio to plugin directory

Closes https://github.com/intake/intake-dremio/issues/9

### CODE REVIEW

**Comment by martindurant** [comment]
This one should be merged before #710 , correct?

If you like this kind of work, intake-parquet and intake-xarray are the most important friend packages, and could do with checking over. They are both considerably simpler.

File: docs/source/api_base.rst [hunk 1]
```diff
@@ -42,9 +41,6 @@ This is a reference API class listing, useful mainly for developers.
 .. autoclass:: intake.source.cache.BaseCache
    :members:
 
-.. autoclass:: intake.source.base.AliasSource
```
Comment (martindurant) [inline docs/source/api_base.rst:45]: moved to intake.source.derived.AliasSource

File: docs/source/api_other.rst [hunk 2]
```diff
@@ -131,15 +129,8 @@ GUI
 .. autoclass:: intake.interface.catalog.select.CatSelector
    :members:
 
-.. autoclass:: intake.interface.source.defined_plots.Plots
```
Comment (martindurant) [inline docs/source/api_other.rst:134]: These both still exist. Perhaps they need panel in the environment to import successfully?

File: docs/source/api_user.rst [hunk 3]
```diff
@@ -7,9 +7,7 @@ These are reference class and function definitions likely to be useful to everyo
    intake.open_catalog
    intake.registry
    intake.register_driver
-   intake.unregister_driver
```
Comment (martindurant) [inline docs/source/api_user.rst:10]: intake.source.discovery.unregister_driver ; should be intake.unregister_driver, but that points incorrectly to intake.source.unregister_driver.

File: docs/source/making-plugins.rst [hunk 4]
```diff
@@ -343,9 +343,9 @@ no entrypoints, it will issue a ``FutureWarning``.
 Python API to Driver Discovery
 ''''''''''''''''''''''''''''''
 
-.. autofunction:: intake.source.discovery.autodiscover
-.. autofunction:: intake.source.discovery.enable
-.. autofunction:: intake.source.discovery.disable
+.. autofunction:: intake.source.discovery.DriverSouces.register_driver
```
Comment (martindurant) [inline docs/source/making-plugins.rst:346]: intake.source.discover.drivers.register_driver (`drivers` is an import-time instance of DriverSources)

File: docs/source/transforms.rst [hunk 5]
```diff
@@ -164,23 +164,20 @@ Plugin examples
...
- 
+
 .. _xarray-plugin-transform: https://github.com/intake/intake-xarray/blob/master/intake_xarray/derived.py#L38
 
 API
 ```
 
 .. autosummary::
    intake.source.derived.DerivedSource
-   intake.source.derived.Alias
```
Comment (martindurant) [inline docs/source/transforms.rst:175]: intake.source.derived.AliasSource

**Review by martindurant — COMMENTED** [review]
Thanks for the digging here!
I updated a few misses. Sorry for moving things around - a docs CI run would definitely have found these.

**Comment by blakerosenthal** [comment]
> This one should be merged before #710 , correct?
> 
> If you like this kind of work, intake-parquet and intake-xarray are the most important friend packages, and could do with checking over. They are both considerably simpler.

Yep, should be a lot easier to merge this one first. Docs look pretty good for both those friend packages (or at least building properly), or are you referring to the work in #710?

Thanks for all the notes! I've added everything back in with updated namespaces.

**Comment by martindurant** [comment]
> Docs look pretty good for both those friend packages (or at least building properly), or are you referring to the work in https://github.com/intake/intake/pull/710?

I meant both :)

https://intake.github.io/status/ , based on https://github.com/intake/status would be very nice to update (no one uses TravisCI anymore!).

File: docs/source/api_user.rst [hunk 6]
```diff
@@ -9,7 +9,6 @@ These are reference class and function definitions likely to be useful to everyo
    intake.register_driver
    intake.unregister_driver
    intake.upload
-   intake.open_
```
Comment (martindurant) [inline docs/source/api_user.rst:12]: The idea was, that for an example driver like CSVSource, which has name="csv", a function will be generated intake.open_csv - and also for all other named and found drivers. This item was supposed to document this behaviour, but there is of course no function just called open_. Perhaps the API docs is the wrong place to put it and you are right to simply remove it here.

Comment (blakerosenthal) [reply docs/source/api_user.rst:12]: Sphinx might have had an update or something recently, but it was very unhappy about `intake.open_` not being importable. I do think there are better places for it, but I wish Sphinx could understand this better.

**Pull request merged**